### PR TITLE
add a configuration to control columns visibility and width

### PR DESCRIPTION
**Pull request by dlax**
This is an alternative to #370 which adds support for configuring pg_activity from a file. ~~The JSON format is to be discussed.~~ (EDIT: changed to INI)

What do you think @blogh?

### CODE REVIEW

**Comment by dlax** [comment]
> The JSON format is to be discussed.

We can also use an INI format, if we anticipate that we'll never need nested options (or sub-sections).

**Comment by blogh** [comment]
At first, I was not fond of the idea of  a `json` file, but it's not that bad. I think we can keep this format.

Do we remove these options from the cli ?

```
Process table display options:
  These options may be used hide some columns from the processes table.

  --no-pid              Disable PID.
  --no-database         Disable DATABASE.
  --no-user             Disable USER.
  --no-client           Disable CLIENT.
  --no-cpu              Disable CPU%.
  --no-mem              Disable MEM%.
  --no-read             Disable READ/s.
  --no-write            Disable WRITE/s.
  --no-time             Disable TIME+.
  --no-wait             Disable W.
  --no-app-name         Disable App.
```

**Comment by dlax** [comment]
> At first, I was not fond of the idea of a |json| file, but it's not that 
> bad. I think we can keep this format.

Not very happy with JSON either; it's just the simplest solution.

But will think about other options and try to see how it works.

> Do we remove these options from the cli ?

I don't think it's needed for a breaking change. As is, command-line 
options take precedence over the configuration file so this is properly 
handled. Also, from a user point of view, I suspect people would want to 
sometime hide a specific column one-shot through the --no-xxx option, 
but probably won't change columns width so often.

**Comment by blogh** [comment]
That makes sens, yes.

**Comment by dlax** [comment]
Changed to INI format; now ready for testing and review @blogh.

File: README.md [hunk 1]
```diff
@@ -138,6 +138,23 @@ ex:
       --no-proc-info        Display workers process information in header.
       --refresh REFRESH     Refresh rate. Values: 0.5, 1, 2, 3, 4, 5 (default: 2).
 
+## Configuration
+
+`pg_activity` may be configured through configuration file, in [INI format][],
+read from `${XDG_CONFIG_HOME:~/.config}/pg_activity.conf` or
+`/etc/pg_activity.conf` in that order. Command-line options may override
```
Comment (blogh) [inline README.md:145]: Should we specify that it's exclusive ?

**Review by blogh — CHANGES_REQUESTED** [review]
How would it work if we add a panel with different columns (I suppose it would work, we already have mem/cpu/read/write)?

Does it make sense to hide columns in some panels but not in others, should we support it? (eg displaying client only for blocking queries)

File: README.md (hunk 1 above)
Comment (dlax) [reply README.md:145]: It's not "exclusive"; rather command-line options take precedence.

**Comment by dlax** [comment]
> How would it work if we add a panel with different columns (I suppose it 
> would we already have mem/cpu/read/write)?

It works the same as before: if a column is configured to be hidden, it 
will be hidden in all panels.

> Does it make sense to hide columns in some panels but not in others? (eg 
> displaying client only for blocking queries)

I don't know, maybe... but maybe not...
Yet, if we want a way to configure per-panel visibility, we'd need a way 
to select the panel name from the configuration file. Since there is no 
possibility to nest options in INI format, the nesting should be in the 
section mame, i.e. something like `[running.client]`. Or we should 
rather go with a file format supporting nested data structures: but is 
it worth?

File: README.md (hunk 1 above)
Comment (blogh) [reply README.md:145]: I mean it's `/etc/pg_activity.conf` OR `~user/.config/pg_activity.conf`.

Comment (dlax) [reply README.md:145]: That's how it's written already: "read from `${XDG_CONFIG_HOME:~/.config}/pg_activity.conf` or `/etc/pg_activity.conf`". Seems explicit enough to me.

**Comment by dlax** [comment]
I'm in favor of the INI format. It's certainly simpler for the user and
the complexity on our side, arguably a bit more important than JSON (but
without the need for a third-party dependency), is acceptable.

File: README.md (hunk 1 above)
Comment (blogh) [reply README.md:145]: I first understood it as either. But it's probably my English acting up.

**Comment by blogh** [comment]
Ok, let commit to this then. Thanks !

**Pull request merged**